src/prefect/server/concurrency/lease_storage/__init__.py

```diff
@@ -15,9 +16,22 @@ class ConcurrencyLeaseStorageModule(Protocol):
     ConcurrencyLeaseStorage: type[ConcurrencyLeaseStorage]
 
 
-@dataclass
-class ConcurrencyLimitLeaseMetadata:
+class ConcurrencyLeaseHolder(BaseModel):
+    """Model for validating concurrency lease holder information."""
+
+    model_config: ClassVar[ConfigDict] = ConfigDict(extra="forbid")
+
+    type: Literal["flow_run", "task_run", "deployment"]
+    id: UUID
+
+
+class ConcurrencyLimitLeaseMetadata(BaseModel):
+    """Model for validating concurrency limit lease metadata."""
+
+    model_config: ClassVar[ConfigDict] = ConfigDict(extra="forbid")
+
     slots: int
+    holder: Optional[dict[str, Any]] = None
```

Comment:
Should we use the model here?
```suggestion
    holder: Optional[ConcurrencyLeaseHolder] = None
```